src/lib.rs

```diff
@@ -51,3 +53,9 @@ pub mod vm;
 pub use crate::plan::{
     AllocationSemantics, BarrierSelector, Mutator, MutatorContext, ObjectQueue, Plan,
 };
+
+static NUM_CONCURRENT_TRACING_PACKETS: AtomicUsize = AtomicUsize::new(0);
```

Comment:
I think this should be moved into `GlobalState`. Whoever picks up this PR can do that refactoring. I think we should reduce our reliance on statics in general.

Reply:
This field is now removed. We now check if the concurrent bucket is drained, or is empty.